rl/wolf_sheep/agents.py

```diff
@@ -0,0 +1,78 @@
+from mesa_models.wolf_sheep.agents import GrassPatch, Sheep, Wolf
+
+from .utility import move
+
+
+class SheepRL(Sheep):
+    def step(self):
+        """
+        The code is exactly same as mesa-example with the only difference being the move function and new sheep creation class.
+        """
+        action = self.model.action_dict[self.unique_id]
+        move(self, action)
+
+        living = True
+
+        if self.model.grass:
+            # Reduce energy
+            self.energy -= 1
+
+            # If there is grass available, eat it
+            this_cell = self.model.grid.get_cell_list_contents([self.pos])
+            grass_patch = next(obj for obj in this_cell if isinstance(obj, GrassPatch))
+            if grass_patch.fully_grown:
+                self.energy += self.model.sheep_gain_from_food
+                grass_patch.fully_grown = False
+
+            # Death
+            if self.energy < 0:
+                self.model.grid.remove_agent(self)
+                self.model.schedule.remove(self)
+                living = False
+
+        if living and self.random.random() < self.model.sheep_reproduce:
+            # Create a new sheep:
+            if self.model.grass:
+                self.energy /= 2
+            unique_id_str = f"sheep_{self.model.next_id()}"
+            lamb = SheepRL(unique_id_str, self.pos, self.model, self.moore, self.energy)
+            self.model.grid.place_agent(lamb, self.pos)
+            self.model.schedule.add(lamb)
+
+
+class WolfRL(Wolf):
```

Comment:
Do you need to inherit this from mesa_models?  My concern is any change to that model will break this model and that inheriting these parameter is not necessary so it makes it unnecessarily brittle.